pkg/users/local.js

```diff
@@ -37,18 +38,79 @@ function AccountsPage() {
     const groups = useFile("/etc/group", { syntax: etc_group_syntax });
     const current_user_info = useLoggedInUser();
 
-    if (!accounts || !groups || !current_user_info)
+    const [details, setDetails] = useState(null);
+    useEffect(() => {
+        getLogins().then(setDetails);
+
+        // Watch `/var/run/utmp` to register when user logs in or out
+        const handle = cockpit.file("/var/run/utmp", { superuser: "try", binary: true });
+        handle.watch(() => {
+            getLogins().then(setDetails);
+        });
+        return handle.close;
+    }, [path, accounts, shadow]);
+
+    if (!accounts || !groups || !current_user_info || !details)
         return null;
 
-    if (path.length === 0)
-        return <AccountsList accounts={accounts} current_user={current_user_info.name} />;
-    else
+    // lastlog uses same sorting as /etc/passwd therefore arrays can be combined based on index
+    const accountsInfo = accounts.map((account, i) => {
+        return Object.assign({}, account, details[i]);
+    });
+
+    if (path.length === 0) {
+        return <AccountsMain accountsInfo={accountsInfo} current_user={current_user_info.name} groups={groups} />;
+    } else
         return (
-            <AccountDetails accounts={accounts} groups={groups} shadow={shadow}
+            <AccountDetails accounts={accountsInfo} groups={groups} shadow={shadow}
                             current_user={current_user_info.name} user={path[0]} />
         );
 }
 
+function get_locked(name) {
+    return cockpit.spawn(["/usr/bin/passwd", "-S", name], { environ: ["LC_ALL=C"], superuser: "require" })
+            .catch(() => "")
+            .then(content => {
+                const status = content.split(" ")[1];
+                // libuser uses "LK", shadow-utils use "L".
+                return status && (status == "LK" || status == "L");
+            });
+}
+
+async function getLogins() {
+    const lastlog = await cockpit.spawn(["/usr/bin/lastlog"], { environ: ["LC_ALL=C"] })
+            .catch(err => console.warn("Unexpected error when getting last login information", err));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17830-20221201-153346-bf483bf9-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)